api/visibility/visibility_controller.go

```diff
@@ -111,13 +146,23 @@ func (c *Controller) patchVisibility(r *web.Request) (*web.Response, error) {
 	ctx := r.Context()
 	log.C(ctx).Debugf("Updating visibility  with id %s", visibilityID)
 
-	visibility, err := c.VisibilityStorage.Get(ctx, visibilityID)
+	visibility, err := c.Repository.Visibility().Get(ctx, visibilityID)
 	if err != nil {
-		return nil, util.HandleStorageError(err, "visibility", visibilityID)
+		return nil, util.HandleStorageError(err, "visibility")
 	}
 
 	createdAt := visibility.CreatedAt
 
+	changes, err := query.LabelChangesFromJSON(r.Body)
```

Comment:
why unmarshal the labels and the visibility itself in two parts ?

Reply:
because that's not labels, but label changes